### DISCUSSION TITLE
[Question] Can the offset directive be used to extend a nodes range beyond the match

### DISCUSSION THREAD

**Discussion by akinsho** [nvim-treesitter/nvim-treesitter#3124] [General]
Hi, 

Apologies for this question as issue, but possible an answer might mean that users aren't asking the same question in the future. So I'm trying to find out if I can use the `#offset!` directive to make a range highlights extend beyond the actual match or is mainly intended for truncating a range .e.g.

I have comment highlights that look like below.
 
<img width="130" alt="image" src="https://user-images.githubusercontent.com/22454918/177573848-76e351d1-a340-47d6-9292-b56c8cddc61e.png">

What I'm experimenting with is extending the start of the range to be one character before the start of the name .e.g instead of `FIXME:` I'd like ` FIXME:` or to make it clear I'll make the space character an `x` and say instead of `FIXME:` I'd like to match `xFIXME:` using a directive like
`(#offset! 0 -1 0 1)` to "extend" the range of the match out by one on the left and by one on the right. Is this possible?

**Comment by kyazdani42** [comment]
I think this was upstreamed to neovim itself ? see `:help ts-directive-offset!`

**Reply by kyazdani42** [reply]
But i'm not sure you would be able to match outside of the node range.

**Reply by akinsho** [reply]
Thanks, @kyazdani42. I've tried it with `0 -1 0 -1` but this doesn't seem to move it outside the range as far as I can tell but there are lots of issues in this repo that use negative values which made me think it was possible to move out of bounds
